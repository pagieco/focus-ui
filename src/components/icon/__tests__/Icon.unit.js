import { shallowMount } from '@vue/test-utils';
import { Icon } from '../index';

const shallowWrap = (propsData = {}) => shallowMount(Icon, { propsData });

describe('Icon', () => {
  it('is a valid vue component', () => {
    expect(Icon).toBeAComponent();
  });

  it('matches the snapshot', () => {
    const propsData = { name: 'user' };

    expect(shallowMount(Icon, { propsData }).element).toMatchSnapshot();
  });

  it('uses the name prop', () => {
    const name = 'times';
    const wrapper = shallowWrap({ name });

    expect(wrapper.attributes('icon')).toBe(name);
  });

  it('use the accessibility-label prop', () => {
    const accessibilityLabel = 'Circle plus icon';
    const wrapper = shallowWrap({ accessibilityLabel, name: 'plus-circle' });

    expect(wrapper.attributes('aria-label')).toBe(accessibilityLabel);
  });
});
