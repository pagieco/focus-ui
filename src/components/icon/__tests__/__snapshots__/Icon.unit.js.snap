// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon matches the snapshot 1`] = `
<fontawesomeicon-stub
  icon="user"
/>
`;
